service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/aws/s3StorageFolder/ControlledAwsS3StorageFolderResourceTest.java

```diff
@@ -0,0 +1,73 @@
+package bio.terra.workspace.service.resource.controlled.cloud.aws.s3StorageFolder;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.common.exception.MissingRequiredFieldException;
+import bio.terra.workspace.common.BaseAwsUnitTest;
+import bio.terra.workspace.common.fixtures.ControlledAwsResourceFixtures;
+import bio.terra.workspace.common.fixtures.ControlledResourceFixtures;
+import bio.terra.workspace.service.resource.exception.InvalidNameException;
+import org.junit.jupiter.api.Test;
+
+public class ControlledAwsS3StorageFolderResourceTest extends BaseAwsUnitTest {
+
+  @Test
+  public void validateResourceTest() {
```

Comment:
nit: given that these are unit tests with ~0 setup required, prefer splitting them into separate tests

Reply:
Retaining as a single test for now - with more creation param we may need updates later